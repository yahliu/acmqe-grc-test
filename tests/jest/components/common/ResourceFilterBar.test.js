/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import React from 'react'
import ResourceFilterBar from '../../../../src-web/components/common/ResourceFilterBar'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { updateActiveFilters } from '../../../../src-web/actions/common'

describe('ResourceFilterBar', () => {
  const history = {
    'length': 5,
    'action': 'POP',
    'location': {
      'pathname': '/multicloud/policies/',
    }
  }
  const location = {
    'pathname': '/multicloud/policies/'
  }
  it('no filters applied will return null', () => {
    const preloadedState = {
      resourceToolbar: {
        activeFilters: {}
      }
    }
    const mockStore = configureMockStore()
    const storePolicyCluster = mockStore(preloadedState)
    const component = renderer.create(
      <Provider store={storePolicyCluster}>
        <BrowserRouter>
          <ResourceFilterBar
            activeFilters={{}}
            history={history}
            location={location}
            updateActiveFilters={updateActiveFilters}
          />
        </BrowserRouter>
      </Provider>
    )
    expect(component.toJSON()).toBeNull()
  })
  it('w/ HIPAA filter returns single filter chip and clear all link', () => {
    const standardSet = { standards: new Set(['HIPAA']) }
    const activeFilters = {
      ...standardSet,
    }
    const preloadedState = {
      resourceToolbar: {
        activeFilters
      }
    }
    const mockStore = configureMockStore()
    const storePolicyCluster = mockStore(preloadedState)
    const component = renderer.create(
      <Provider store={storePolicyCluster}>
        <BrowserRouter>
          <ResourceFilterBar
            activeFilters={activeFilters}
            history={history}
            location={location}
            updateActiveFilters={updateActiveFilters}
          />
        </BrowserRouter>
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
    // Child 1: Standards Filter chip
    // Child 2: "Clear All" link
    // Clear single filter: should fire Redux store action to clear Standards filter and also clear filter object
    // component.toJSON().children[1].children[1].props.onClick()
    // expect(storePolicyCluster.getActions().length).toEqual(1)
    // expect(storePolicyCluster.getActions()[0]).toEqual({ type: 'ACTIVE_FILTER_UPDATE', activeFilters: {} } )
    // // Click "Clear All": should fire second Redux store action clearing all filters
    // component.toJSON().children[2].props.onClick()
    // expect(storePolicyCluster.getActions().length).toEqual(2)
    // expect(storePolicyCluster.getActions()[1]).toEqual({ type: 'ACTIVE_FILTER_UPDATE', activeFilters: {} })
    // // Key press on "Clear All": should fire third Redux store action clearing all filters
    // component.toJSON().children[2].props.onKeyPress()
    // expect(storePolicyCluster.getActions().length).toEqual(3)
    // expect(storePolicyCluster.getActions()[2]).toEqual({ type: 'ACTIVE_FILTER_UPDATE', activeFilters: {} })
  })
  it('w/ HIPAA filter returns multiple filter chips and clear all link', () => {
    const standardSet = { standards: new Set(['HIPAA']) }
    const typeSet = { type: new Set(['Enforce']) }
    const activeFilters = {
      ...standardSet,
      ...typeSet
    }
    const preloadedState = {
      resourceToolbar: {
        activeFilters
      }
    }
    const mockStore = configureMockStore()
    const storePolicyCluster = mockStore(preloadedState)
    const component = renderer.create(
      <Provider store={storePolicyCluster}>
        <BrowserRouter>
          <ResourceFilterBar
            activeFilters={activeFilters}
            history={history}
            location={location}
            updateActiveFilters={updateActiveFilters}
          />
        </BrowserRouter>
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
    // Child 1: Standards Filter chip
    // Child 2: Enforce Filter chip
    // Child 3: "Clear All" link
    // Clear single filter: should fire Redux store action to clear Standards filter, leaving Types
    // component.toJSON().children[1].children[1].props.onClick()
    // expect(storePolicyCluster.getActions().length).toEqual(1)
    // expect(storePolicyCluster.getActions()[0]).toEqual({ type: 'ACTIVE_FILTER_UPDATE', activeFilters: typeSet } )
    // // Clear single filter: should fire Redux store action to clear Standards filter, leaving Types
    // component.toJSON().children[2].children[1].props.onClick()
    // expect(storePolicyCluster.getActions().length).toEqual(2)
    // expect(storePolicyCluster.getActions()[1]).toEqual({ type: 'ACTIVE_FILTER_UPDATE', activeFilters: standardSet } )
    // // Click "Clear All": should fire second Redux store action clearing all filters
    // component.toJSON().children[3].props.onClick()
    // expect(storePolicyCluster.getActions().length).toEqual(3)
    // expect(storePolicyCluster.getActions()[2]).toEqual({ type: 'ACTIVE_FILTER_UPDATE', activeFilters: {} })
  })
})
