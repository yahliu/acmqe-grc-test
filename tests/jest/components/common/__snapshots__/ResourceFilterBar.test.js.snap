// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceFilterBar w/ HIPAA filter returns multiple filter chips and clear all link 1`] = `
<div
  className="resource-filter-bar"
>
  <span
    className="title"
  >
    Filters:
  </span>
  <div
    className="pf-c-chip"
    data-ouia-component-id="OUIA-Generated-Chip-2"
    data-ouia-component-type="PF4/Chip"
    data-ouia-safe={true}
  >
    <span
      className="pf-c-chip__text"
      id="pf-random-id-1"
    >
      HIPAA
    </span>
    <button
      aria-disabled={false}
      aria-label="close"
      aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
      className="pf-c-button pf-m-plain"
      data-ouia-component-id="close"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe={true}
      disabled={false}
      id="remove_pf-random-id-1"
      onClick={[Function]}
      role={null}
      type="button"
    >
      <svg
        aria-hidden="true"
        aria-labelledby={null}
        fill="currentColor"
        height="1em"
        role="img"
        style={
          Object {
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 352 512"
        width="1em"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </button>
  </div>
  <div
    className="pf-c-chip"
    data-ouia-component-id="OUIA-Generated-Chip-3"
    data-ouia-component-type="PF4/Chip"
    data-ouia-safe={true}
  >
    <span
      className="pf-c-chip__text"
      id="pf-random-id-2"
    >
      Enforce
    </span>
    <button
      aria-disabled={false}
      aria-label="close"
      aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
      className="pf-c-button pf-m-plain"
      data-ouia-component-id="close"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe={true}
      disabled={false}
      id="remove_pf-random-id-2"
      onClick={[Function]}
      role={null}
      type="button"
    >
      <svg
        aria-hidden="true"
        aria-labelledby={null}
        fill="currentColor"
        height="1em"
        role="img"
        style={
          Object {
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 352 512"
        width="1em"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`ResourceFilterBar w/ HIPAA filter returns single filter chip and clear all link 1`] = `
<div
  className="resource-filter-bar"
>
  <span
    className="title"
  >
    Filters:
  </span>
  <div
    className="pf-c-chip"
    data-ouia-component-id="OUIA-Generated-Chip-1"
    data-ouia-component-type="PF4/Chip"
    data-ouia-safe={true}
  >
    <span
      className="pf-c-chip__text"
      id="pf-random-id-0"
    >
      HIPAA
    </span>
    <button
      aria-disabled={false}
      aria-label="close"
      aria-labelledby="remove_pf-random-id-0 pf-random-id-0"
      className="pf-c-button pf-m-plain"
      data-ouia-component-id="close"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe={true}
      disabled={false}
      id="remove_pf-random-id-0"
      onClick={[Function]}
      role={null}
      type="button"
    >
      <svg
        aria-hidden="true"
        aria-labelledby={null}
        fill="currentColor"
        height="1em"
        role="img"
        style={
          Object {
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 352 512"
        width="1em"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </button>
  </div>
</div>
`;
